e2e/specs/register-new-patient.spec.ts

```diff
@@ -6,55 +6,54 @@ import { deletePatient, getPatient } from '../commands';
-
-      await expect(person.display).toBe(`${givenName} ${middleName} ${familyName}`);
-      await expect(person.gender).toMatch(new RegExp(sex[0], 'i'));
-      await expect(dayjs(person.birthdate).format('DD/MM/YYYY')).toBe(formValues.birthdate);
-      await expect(person.preferredAddress.address1).toBe(formValues.address1);
-      await expect(person.preferredAddress.cityVillage).toBe(formValues.cityVillage);
-      await expect(person.preferredAddress.stateProvince).toBe(formValues.stateProvince);
-      await expect(person.preferredAddress.country).toBe(formValues.country);
-      await expect(person.attributes[0].display).toBe(formValues.phone);
-    });
+test('Register a new patient', async ({ page, api }) => {
+  test.setTimeout(5 * 60 * 1000);
+  const patientRegistrationPage = new RegistrationAndEditPage(page);
+
+  // TODO: Add email field after fixing O3-1883 (https://issues.openmrs.org/browse/O3-1883)
+  const formValues: PatientRegistrationFormValues = {
+    givenName: `Johnny`,
+    middleName: 'Donny',
+    familyName: `Ronny`,
+    sex: 'male',
+    birthdate: { day: '01', month: '02', year: '2020' },
+    postalCode: '',
+    address1: 'Bom Jesus Street',
+    address2: '',
+    country: 'Brazil',
+    stateProvince: 'Pernambuco',
+    cityVillage: 'Recife',
+    phone: '5555551234',
+  };
+
+  await test.step('When I visit the registration page', async () => {
+    await patientRegistrationPage.goto();
+    await patientRegistrationPage.waitUntilTheFormIsLoaded();
+  });
+
+  await test.step('And then I click on fill new values into the registration form and then click the `Submit` button', async () => {
+    await patientRegistrationPage.fillPatientRegistrationForm(formValues);
+  });
+
+  await test.step("Then I should be redirected to the new patient's chart page and a new patient record should be created from the information captured in the form", async () => {
+    await expect(page).toHaveURL(new RegExp('^[\\w\\d:\\/.-]+\\/patient\\/[\\w\\d-]+\\/chart\\/.*$'));
+    const patientUuid = /patient\/(.+)\/chart/.exec(page.url())?.[1] ?? null;
+    expect(patientUuid).not.toBeNull();
+
+    const patient = await getPatient(api, patientUuid);
+    const { person } = patient;
+    const { givenName, middleName, familyName, sex } = formValues;
+
+    expect(person.display).toBe(`${givenName} ${middleName} ${familyName}`);
+    expect(person.gender).toMatch(new RegExp(sex[0], 'i'));
+    expect(dayjs(person.birthdate).format('DD/MM/YYYY')).toBe(
+      `${formValues.birthdate.day}/${formValues.birthdate.month}/${formValues.birthdate.year}`,
+    );
+    expect(person.preferredAddress.address1).toBe(formValues.address1);
+    expect(person.preferredAddress.cityVillage).toBe(formValues.cityVillage);
+    expect(person.preferredAddress.stateProvince).toBe(formValues.stateProvince);
+    expect(person.preferredAddress.country).toBe(formValues.country);
+    expect(person.attributes[0].display).toBe(formValues.phone);
```

Comment:
Playwright functions are asynchronous. so, using await ensures that each promise resolves before moving on to the next line of code.

Reply:
Aaah I see, I checked the [Playwright docs](https://playwright.dev/docs/test-assertions) and I thought we were using the [non-retrying assertions](https://playwright.dev/docs/test-assertions#non-retrying-assertions) here that didn't need an await.